### ISSUE TITLE
Add an option to not espace quotes.

### ISSUE THREAD

**Issue by philipwalton** [markedjs/marked#269]
I stumbled upon an issue that's giving me problems because marked is escaping double and single quotes in all situations and there's no option to change it.

https://github.com/chjj/marked/blob/master/lib/marked.js#L999-L1000

I'm no markdown expert, but I'm converting a project to node that previously used Ruby's [rdiscount](http://dafoster.net/projects/rdiscount/), which does not escape quotes. I do not see any reason (from an HTML perspective) to force escaping of quotes, so I think it makes sense to add an option to prevent this behavior.

If this is something you'd consider, I'd be happy to submit a pull request. Just let me know.

**Comment by ethanmick** [comment]
Just ran into this. :+1:

**Comment by adam-lynch** [comment]
Related: #529

**Comment by chris5marsh** [comment]
Plus a really silly point: `&apos;` isn't understood by IE&, which unfortunately is still on my list of supported browsers. An option to turn off some or all entity conversions?

**Comment by cri5ti** [comment]
The code in question is here: https://github.com/chjj/marked/blob/master/lib/marked.js#L685

Shouldn't we have something like this?
```js
if (this.options.sanitize)
    out += this.renderer.text(escape(this.smartypants(cap[0])));
else
    out += this.renderer.text(this.smartypants(cap[0]));        
```

**Comment by rosshinkley** [comment]
I know I'm late to the party, but I just hit this as well doing something similar with Handlebars + marked and thought I'd throw in my +1.

**Comment by r0hitsharma** [comment]
+1 Faced the same issue (Handlebars + marked)

@cri5ti, your suggestion fixes my issue. Thanks!

**Comment by snollygolly** [comment]
+1 Same issue for me as well.

**Comment by patrickheeney** [comment]
+1 Also ran into this. How is everyone working around this issue? Just applying @cri5ti code change after npm install?

**Comment by caizone** [comment]
+1 it's works for me.

**Comment by matt-** [comment]
If someone would like to submit a PR with passing tests that will help move the issue along.

**Comment by KostyaTretyak** [comment]
You can copy and paste this [my commit](https://github.com/KostyaTretyak/marked/blob/83caf40624bb60c5beca2e424bf9135059a0dbd3/lib/marked.js), its improved performance of `marked` on 30-40%. And then you can to change [this rows](https://github.com/KostyaTretyak/marked/blob/83caf40624bb60c5beca2e424bf9135059a0dbd3/lib/marked.js#L1094-L1095) for avoid escape quotes.

My commit differs just by refactoring the `escape()` function. See [my PR](https://github.com/chjj/marked/pull/975) for `marked`.

**Comment by joshbruce** [comment]
Closing as stale and not the focus. See also #1106, #1216, and #1080.

**Comment by marco-silva0000** [comment]
ls there a way to do this in the current version?

**Comment by r-matsuzaka** [comment]
I have the same issue.

**Comment by UziTech** [comment]
You can create a [custom extension](https://marked.js.org/using_pro#extensions) to render markdown differently

```js
const handlebarsTag = {
  name: 'handlebarsTag',
  level: 'inline',                          // This is an inline-level tokenizer
  start(src) { return src.indexOf('{{'); }, // Hint to Marked.js to stop and check for a match
  tokenizer(src, tokens) {
    const rule = /^\{\{.*?\}\}+/;           // Regex for the complete token, anchor to string start
    const match = rule.exec(src);
    if (match) {
      return {                               // Token to generate
        type: 'handlebarsTag',               // Should match "name" above
        raw: match[0],                       // Text to consume from the source
        text: match[0]                       // Additional custom properties
      };
    }
  },
  renderer(token) {
    return token.text;
  }
};

marked.use({extensions: [handlebarsTag]});
```

**Comment by ValentinVignal** [comment]
> You can create a [custom extension](https://marked.js.org/using_pro#extensions) to render markdown differently

@UziTech  I don't think that is working, I tried your sample and I still get `<p>&#39;</p>` as an output

[![Edit TypeScript Playground Export (forked)](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/s/typescript-playground-export-forked-6d1sui?fontsize=14&hidenavigation=1&theme=dark)

**Comment by UziTech** [comment]
My example only prevents escaping between `{{}}` for handlebars. You will need to change the code of you want to prevent it other places.

**Comment by SachaG** [comment]
It may be obvious (and obviously having this solved in the library itself would be better) but personally I found it easiest to just do:

```
html = marked.parse(s).replaceAll('&#39;', `'`)
```

**Comment by paulrudy** [comment]
An example custom extension based on @UziTech's [example](https://github.com/markedjs/marked/issues/269#issuecomment-1173177983) for anyone wanting to prevent escaping `'` (single quotes):

```js
const singleQuote = {
  name: "singleQuote",
  level: "inline", // This is an inline-level tokenizer
  start(src) {
    return src.indexOf("'");
  }, // Hint to Marked.js to stop and check for a match
  tokenizer(src, tokens) {
    const rule = /^'/; // Regex for the complete token, anchor to string start
    const match = rule.exec(src);
    if (match) {
      return {
        // Token to generate
        type: "singleQuote", // Should match "name" above
        raw: match[0], // Text to consume from the source
        text: match[0], // Additional custom properties
      };
    }
  },
  renderer(token) {
    return token.text;
  },
};

marked.use({ extensions: [singleQuote] });
```

**Comment by devbytyler** [comment]
Getting blocked on this now too, but I am only tokenizing markdown, never rendering it. It's not really clear at what point the escaping happens, but seems like it would ideally happen later in the process when rendering is about to happen,

**Comment by UziTech** [comment]
@devbytyler this should be fixed in the next major version by #3495

Sometimes escaping currently is happening during tokenization.
